contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,309 @@
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: extraNumberOfPeriods,
+                totalRewardAmount: 0,
+                maxRewardPerVote: 0
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit NumberOfPeriodsUpdated(extraNumberOfPeriods);
+    }
+
+    /// @notice Manage the reward per vote of the campaign
+    /// @dev This function should be called after the campaign is created
+    /// @param newMaxRewardPerVote New maximum reward per vote
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function manageRewardPerVote(
+        uint256 newMaxRewardPerVote,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId != 0, "Campaign not created");
+        require(newMaxRewardPerVote > 0, "Invalid reward per vote");
+
+        // Manage the campaign
+        ICampaingRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaingRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: 0,
+                totalRewardAmount: 0,
+                maxRewardPerVote: newMaxRewardPerVote
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit RewardPerVoteUpdated(newMaxRewardPerVote);
+    }
+
+    /// @notice calculate the fee amount and transfer it to the feeCollector
+    /// @param amount Amount to calculate the fee
+    /// @return Amount after fee
+    function _handleFee(uint256 amount) internal returns (uint256) {
+        uint256 feeAmount = (amount * fee) / BASE_FEE;
+
+        // If there is a fee, transfer it to the feeCollector
+        if (feeAmount > 0) {
+            IERC20(rewardToken).transfer(feeCollector, feeAmount);
+            // Return the amount after fee
+            return amount - feeAmount;
+        }
+
+        // If there is no fee, return the original amount
+        return amount;
+    }
+
+    ////////////////////////////////////////////////////
+    /// --- GOVERNANCE && OPERATION
+    ////////////////////////////////////////////////////
+    /// @notice Set the campaign id
+    /// @dev Only callable by the governor or strategist
+    /// @param _campaignId New campaign id
+    function setCampaignId(uint256 _campaignId)
+        external
+        onlyGovernorOrStrategist
+    {
+        campaignId = _campaignId;
+        emit CampaignIdUpdated(_campaignId);
+    }
+
+    /// @notice Rescue ETH from the contract
+    /// @dev Only callable by the governor or strategist
+    /// @param receiver Address to receive the ETH
+    function rescueETH(address receiver) external onlyGovernorOrStrategist {
```

Comment:
Let's add `receiver != address(0)` check here

Reply:
Done in this commit: https://github.com/OriginProtocol/origin-dollar/pull/2327/commits/345cbb92c3199560f269765a3b39dd4ca5854350